components/auth/org.wso2.carbon.auth.core/src/main/java/org/wso2/carbon/auth/core/exception/ExceptionCodes.java

```diff
@@ -36,7 +36,10 @@
 
     //oauth2 grants related exception codes
     OAUTH2_GRANT_PROCESS_EXCEPTION(900500, "Internal OAuth2 grant processing error", 500,
-            "Error while processing OAuth2 grant request");
+            "Error while processing OAuth2 grant request"),
+    INVALID_REQUEST(900600, "Invalid request", 400, "Invalid request"),
+    UNSUPPORTED_SCOPE(900601, "Unsupported scope", 400, "Unsupported scope"),
```

Comment:
Word Unsupported can be misleading, AFAIU there are no unsupported scopes, there can be invalid scopes.

Reply:
Userinfo only supports openid scope. When you provide a token for userinfo, that token's scope should be openid. You can generate tokens with different scopes. But userinfo API will only work for tokens with openid scope. That's why there is an unsupported scope.